### PR TITLE
Improved MapFeatureInfo decorator to display properly highlight geometries when terrain is enabled

### PR DESCRIPTION
**Pull request by mdastous-bentley**
The actual MapFeatureInfo tool has been improved to highlight map-layer geometries when the terrain is turned ON.  

To achieve this, the appropriate resolution terrain tiles needs to be collected from the tile tree, then the `@itwin/core-geometry` functionalities are used to drape the map-layer geometries (which include point, lines and polygons) to be draped on the terrain mesh.  The resulting geometries are displayed as decorations.

In the following screenshot, we can see sample highlighted : 
image

### CODE REVIEW

**Comment by pmconne** [comment]
> The resulting geometries are displayed as decorations

How many geometries are we talking about?

> In the following screenshot, we can see sample highlighted 

This refers specifically to the single brigher blue polygon? Are the other polygons also drawn as decorations, or something else?

**Comment by mdastous-bentley** [comment]
> > The resulting geometries are displayed as decorations
> 
> How many geometries are we talking about?
> 
> > In the following screenshot, we can see sample highlighted
The tool only support clicked point selection, so the general workflow is a single geometry at a time.  You can always zoom out, but then the terrain tile resolution and also the map-layers geometries will be low resolution.


> This refers specifically to the single brigher blue polygon? Are the other polygons also drawn as decorations, or something else?
Yes the brighter blue polygon is a decoration.  Decorations are only used to simulate "feature classification".  Other blue polygons are map-layers textures.

**Comment by mdastous-bentley** [comment]
@aruniverse I understand it might be too late, but would be nice to get this included in 4.6.

**Review by pmconne — APPROVED** [review]

File: extensions/map-layers-formats/src/Tools/GeometryTerrainDraper.ts [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+      }
+    }
+
+    return IndexedPolyfaceSubsetVisitor.createSubsetVisitor(mesh, topFacets, 0);
+  }
+
+  public drapeLineString(outStrings: CurvePrimitive[], inPoints: GrowableXYZArray, tolerance: number, range: Range3d): "loading" | "complete" {
+    const tree = this.treeRef.treeOwner.load();
+    if (!tree)
+      return "loading";
```
Comment (aruniverse) [inline extensions/map-layers-formats/src/Tools/GeometryTerrainDraper.ts:109]: nit: i hate the use of literal strings being used over and over in this file; would prefer pulling these into constants.

**Review by aruniverse — APPROVED** [review]

**Comment by aruniverse** [comment]
@mergifyio backport release/4.6.x

File: extensions/map-layers-formats/src/Tools/GeometryTerrainDraper.ts (hunk 1 above)
Comment (mdastous-bentley) [reply extensions/map-layers-formats/src/Tools/GeometryTerrainDraper.ts:109]: ok will take note of that in future updates

**Comment by mdastous-bentley** [comment]
@eringram please review

**Review by eringram — APPROVED** [review]

File: extensions/map-layers-formats/src/Tools/GeometryTerrainDraper.ts (hunk 1 above)
Comment (mdastous-bentley) [reply extensions/map-layers-formats/src/Tools/GeometryTerrainDraper.ts:109]: @aruniverse can you please resolve

**Pull request merged**